src/bio/bio_bulk.c

```diff
@@ -474,19 +474,23 @@ bulk_get_hdl(struct bio_desc *biod, unsigned int pg_cnt,
 }
 
 static inline bool
-bypass_bulk_cache(struct bio_iov *biov, unsigned int pg_cnt)
+bypass_bulk_cache(struct bio_desc *biod, struct bio_iov *biov,
+		  unsigned int pg_cnt)
 {
 	/* Hole, no RDMA */
 	if (bio_addr_is_hole(&biov->bi_addr))
 		return true;
+	/* Huge IOV, allocate DMA buffer & create bulk handle on-the-fly */
+	if (pg_cnt > bio_chk_sz)
+		return true;
+	/* Get buffer operation */
+	if (biod->bd_type == BIO_IOD_TYPE_GETBUF)
```

Comment:
Why ignore BIO_IOD_TYPE_GETBUF for "Huge IOV" case ?

Reply:
No matter what the request is for (update, fetch or getbuf), we don't use bulk cache for "huge iov".